core/controllers/admin.py

```diff
@@ -2345,3 +2345,49 @@ def put(self) -> None:
                 language_coordinator, language_id)
 
         self.render_json({})
+
+
+class InteractionsByExplorationIdHandlerNormalizedRequestDict(TypedDict):
+    """Dict representation of InteractionsByExplorationIdHandler's
+    normalized_request dictionary.
+    """
+
+    exp_id: str
+
+
+class InteractionsByExplorationIdHandler(
+    base.BaseHandler[
+        InteractionsByExplorationIdHandlerNormalizedRequestDict, Dict[str, str]
+    ]
+):
+    """Handler for admin to retrive the list of interactions used in
+    an exploration.
+    """
+
+    GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
+    URL_PATH_ARGS_SCHEMAS: Dict[str, str] = {}
+    HANDLER_ARGS_SCHEMAS = {
+        'GET': {
+            'exp_id': {
+                'schema': {
+                    'type': 'basestring'
+                }
+            }
+        }
+    }
+
+    @acl_decorators.can_access_admin_page
+    def get(self) -> None:
+        assert self.normalized_request is not None
+        exploration_id = self.normalized_request['exp_id']
+
+        exploration = exp_fetchers.get_exploration_by_id(
+            exploration_id, strict=False)
+        if exploration is None:
+            raise self.InvalidInputException('Exploration does not exist.')
+
+        interaction_ids = {
+            state.interaction.id for state in
+            exploration.states.values() if state.interaction.id is not None
+        }
+        self.render_json({'interactions': list(interaction_ids)})
```

Comment:
I agree, Done.

Reply:
Just a note -- if we wanted to do that, we should actually be doing it using a domain object, not a dict. The problem with using raw dicts is that the types become unclear if they get modified in the middle of the method, and this can result in bugs.

Also now the naming interaction_ids is actually incorrect -- one would expect it to be a list of strings and not a list of dicts, just based on the name.

@AFZL210 Would you be OK changing this back to a list of strings for the time being? Also I don't think you need the cast here since it's already a list.

Reply:
Make sense, we can create a domain object in the future if we plan to send other data too, and for now I will change it to a simple list of strings.

Reply:
(For posterity: the update was made in https://github.com/oppia/oppia/pull/20020 .)